components/ExerciseForm/ExerciseForm.tsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import {KeyboardTypeOptions, ScrollView, View} from 'react-native';
+import {exercises} from '../../utils/constants';
+import {
+  ButtonType,
+  ExerciseDetailsForm,
+  ExerciseFormProps,
+  Fields,
+  FormField,
+} from '../../types/CommonTypes';
+import Form from '../Form/Form';
+
+import globalStyle from '../../assets/styles/globalStyle';
+
+const ExerciseForm: React.FC<ExerciseFormProps> = ({
+  submitButtonTitle,
+  onPressSubmit,
+  initialExerciseValues,
+}) => {
+  const type: ButtonType = 'submit';
+  const keyboardType: KeyboardTypeOptions = 'numeric';
+  const exerciseFields: Array<FormField<ExerciseDetailsForm>> = [
+    {
+      label: exercises.EXERCISE_NAME,
+      name: Fields.name,
+      required: true,
+    },
+    {
+      label: exercises.TIMES,
+      keyboardType,
+      name: Fields.times,
+      required: false,
+    },
+    {
+      label: exercises.DURATION,
+      keyboardType,
+      name: Fields.duration,
+      required: true,
+    },
+    {
+      label: exercises.DESCRIPTION,
+      name: Fields.description,
+      multiline: true,
+      numberOfLines: exercises.DEFAULT_NUMBER_OF_TEXTAREA,
+      required: false,
+    },
+  ];
+
+  const exerciseButtons = [
+    {
+      title: submitButtonTitle,
+      type,
+      onPress: (exerciseDetails?: ExerciseDetailsForm) =>
+        onPressSubmit(exerciseDetails),
+    },
+  ];
+
+  return (
+    <ScrollView
+      showsVerticalScrollIndicator={false}
+      contentContainerStyle={globalStyle.pageContainer}>
+      <View style={[globalStyle.flex]}>
+        <Form<ExerciseDetailsForm>
+          fields={exerciseFields}
+          buttons={exerciseButtons}
```

Comment:
In almost every screen that you pass `buttons` to your 'Form' component, they are an array with only a single item, which probably indicates wrong data structure. Maybe this should have been an object.